src/Database/DBA.php

```diff
@@ -467,6 +467,47 @@ public static function count($table, array $condition = [])
 		return self::getClass(Database::class)->count($table, $condition);
 	}
 
+	/**
+	 * Build the table query substring from one or more tables, with or without a schema.
+	 *
+	 * Expected formats:
+	 * - table
+	 * - [table1, table2, ...]
+	 * - [schema1 => table1, schema2 => table2, table3, ...]
+	 *
+	 * @param string|array $tables
+	 * @return string
+	 */
+	public static function buildTableString($tables)
+	{
+		if (is_string($tables)) {
+			$tables = [$tables];
+		}
+
+		$quotedTables = [];
+
+		foreach ($tables as $schema => $table) {
+			if (is_numeric($schema)) {
+				$quotedTables[] = self::quoteIdentifier($table);
+			} else {
+				$quotedTables[] = self::quoteIdentifier($schema) . '.' . self::quoteIdentifier($table);
+			}
+		}
+
+		return implode(', ', $quotedTables);
+	}
+
+	/**
+	 * Escape an identifier (table or field name)
+	 *
+	 * @param $identifier
+	 * @return string
+	 */
+	public static function quoteIdentifier($identifier)
```

Comment:
Wouldn't it be better if this function would be part of ```Database.php```? The reason I'm asking is because this function would be different with - for example - PostgreSQL.

Reply:
We don't have PostgreSQL support right now, so I'm following the current database class/method architecture, where utility methods not needing an open database connection are in `DBA`. Since it has no dependencies, it will be easily moved to the MySQL adapter once someone™ adds PostgreSQL support.

Reply:
I just asked because I'm partially unsure about it. Hopefully @nupplaphil can enlight us?

Reply:
Unsure about what? PostgreSQL support or the class architecture?

Reply:
No, unsure about where to place this function. From my point of view this PR can be merged. But @nupplaphil had got something in mind when creating this structure and I don't want to violate this.

Reply:
From my perspective, the change is OK. I'm currently unsure how we will add the PostgreSQL support and currently, all static methods without db connections stay inside `DBA`, so I don't see a reason to move it to the `Database` class :-)